Quantum-Control-Applications-QuAM/Superconducting/configuration/generate_superconducting_quam.py

```diff
@@ -0,0 +1,312 @@
+import math
+import os.path
+from typing import Dict
+from quam.components import pulses, Octave, IQChannel, DigitalOutputChannel
+
+from quam_components import Transmon, ReadoutResonator, QuAM, FluxLine
+from qualang_tools.units import unit
+from pathlib import Path
+import json
+
+# Class containing tools to help handling units and conversions.
+u = unit(coerce_to_integer=True)
+
+
+def create_default_wiring(num_qubits: int) -> dict:
+    """
+    Create a wiring config tailored to the number of qubits.
+
+    """
+    wiring = {"qubits": {}}
+
+    # Generate example wiring by default
+    xy_lines, flux_lines, res_lines = _example_wiring(num_qubits)
+
+    # Uncomment this to manually set wiring for each qubit
+    # res_lines  = [("con1", 1, "octave1", 1)]*num_qubits
```

Comment:
My own preference would be to group these lines per qubit, e.g. something like
```
qubit_lines = [
    {"xy": ("con1", 1), "readout": ...}
]
```
Any reason for grouping it the other way?

Reply:
The default allocation I made groups channels first by resonator lines, then by XY lines, then by flux lines. This provides the most convenient physical wiring between the Octave and OPX+ for example imagine this at 5-qubits
![wiring-scheme](https://github.com/qua-platform/qua-libs/assets/38335668/b87b821d-09cc-40a3-a302-20850c21f01f)

Since they are physically grouped by channel "type", they are grouped by channel type in the code. 

Let me know your thoughts- happy to change it.